torch/testing/_internal/common_methods_invocations.py

```diff
@@ -470,6 +470,25 @@ def sample_inputs_index_select(op_info, device, dtype, requires_grad):
                         0, torch.tensor(0, dtype=torch.int64, device=device))),
             )
 
+def sample_inputs_index_fill(op_info, device, dtype, requires_grad):
+    samples = []
+    t = make_tensor((S, S, S), device, dtype,
+                    low=None, high=None,
+                    requires_grad=requires_grad)
+    # non-contiguous input
+    t01 = t.transpose(0, 1)
+    t02 = t.transpose(0, 2)
+    t12 = t.transpose(1, 2)
+    idx = index_variable(1, S, device=device)
+    # non-contiguous index
+    idx_nonctg = torch.empty_strided((S,), (2,), device=device, dtype=dtype).long()
+    idx_nonctg.copy_(idx)
+    for d in range(t.dim()):
+        for tensor in [t, t01, t02, t12]:
+            samples.append(SampleInput((tensor, d, idx, -1)))
```

Comment:
let's also add test for complex fill value

Reply:
Already added through OpInfo.

Reply:
wait I don't see an input with complex tensor and complex scalar value ...

Reply:
also let's update the doc to indicate `val` can be either complex or float https://pytorch.org/docs/stable/tensors.html#torch.Tensor.index_fill_

Reply:
Sorry, complex fill value, my bad.

Reply:
`jit_consistency` test fail when fill value is a complex number of the form `a+jb` with the following message:
```
                                            (*sample.input,) + sample.args,                                                                       
                                            sample.kwargs,                                                                                        
>                                           no_grad=(dtype not in dtypes_to_grad_check))

test/test_ops.py:293:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
torch/testing/_internal/common_jit.py:63: in check_against_reference
    outputs_test = self.runAndSaveRNG(func, nograd_inputs, kwargs)
torch/testing/_internal/common_jit.py:130: in runAndSaveRNG
    results = func(*inputs, **kwargs)
torch/testing/_internal/jit_metaprogramming_utils.py:298: in script_fn
    fn, tensors = gen_script_fn_and_args(method_name, func_type, *args, **kwargs)
torch/testing/_internal/jit_metaprogramming_utils.py:290: in gen_script_fn_and_args
    CU = torch.jit.CompilationUnit(script)
torch/jit/_script.py:1103: in __init__
    self.define(lang, _frames_up=_frames_up + 1)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <torch.jit._script.CompilationUnit object at 0x7f2c1019f630>
lang = '\ndef the_method(i0, i1):\n    return torch.index_fill(i0, 0, i1, (-1+1j))\n'
rcb = <function createResolutionCallbackFromEnv.<locals>.<lambda> at 0x7f2be4c49400>, _frames_up = 1

    def define(self, lang, rcb=None, _frames_up=0):
        if not rcb:
            rcb = _jit_internal.createResolutionCallbackFromFrame(_frames_up + 1)
>       self._c.define(lang, rcb)
E       RuntimeError: expected ) but found 'ident' here:
E         File "<string>", line 3
E       
E       def the_method(i0, i1):
E           return torch.index_fill(i0, 0, i1, (-1+1j))
E                                                   ~ <--- HERE

torch/jit/_script.py:1108: RuntimeError
==========================================
```

Reply:
yeah that's expected. JIT doesn't know how to interpret complex literals right now. you can just skips this test in `OpInfo`, by adding:
```
skips=(
                 SkipInfo('TestCommon', 'test_variant_consistency_jit',
                          device_type='cuda', dtypes=[torch.cfloat, torch.cdouble]),
             )),
```

Reply:
I wrapped the complex scalar into a Tensor, there is an `index_fill` overload that accepts singleton Tensors.